learning/courses/basics-of-quantum-information/entanglement-in-action/qiskit-implementation.ipynb

```diff
@@ -0,0 +1,1588 @@
+    "\n",
+    "#### Deterministic strategies\n",
+    "\n",
+    "We'll start with *deterministic* strategies, where Alice's answer $a$ is a function of the question $x$ that she receives, and likewise Bob's answer $b$ is a function of the question $y$ he receives.\n",
+    "So, for instance, we may write $a(0)$ to represent Alice's answer when her question is $0,$ and $a(1)$ to represent Alice's answer when her question is $1.$\n",
+    "\n",
+    "No deterministic strategy can possibly win the CHSH game every time.\n",
+    "One way to reason this is simply to go one-by-one through all of the possible deterministic strategies and check that every one of them loses for at least one of the four possible question pairs.\n",
+    "Alice and Bob can each choose from four possible functions from one bit to one bit — which we encountered in the lesson on [Single systems](https://learning.quantum.ibm.com/course/basics-of-quantum-information/single-systems) — and so there are $16$ different deterministic strategies in total to check.\n",
+    "\n",
+    "We can also reason this analytically.\n",
+    "If Alice and Bob's strategy wins when $(x,y) = (0,0),$ then it must be that $a(0) = b(0);$\n",
+    "if their strategy wins when $(x,y) = (0,1),$ then $a(0) = b(1);$ and similarly,\n",
+    "if the strategy wins for $(x,y)=(1,0)$ then $a(1) = b(0).$\n",
+    "So, if their strategy wins for all three possibilities, then\n",
+    "\n",
+    "$$\n",
+    "b(1) = a(0) = b(0) = a(1).\n",
+    "$$\n",
+    "\n",
+    "This implies that the strategy loses in the final case $(x,y) = (1,1),$ for here winning requires that\n",
+    "$a(1) \\neq b(1).$\n",
+    "Thus, there can be no deterministic strategy that wins every time.\n",
+    "\n",
+    "On the other hand, it is easy to find deterministic strategies that win in three of the four cases, such as $a(0)=a(1)=b(0)=b(1)=0.$\n",
+    "From this we conclude that the maximum probability for Alice and Bob to win using a deterministic strategy is $3/4.$\n",
+    "\n",
+    "#### Probabilistic strategies\n",
+    "\n",
+    "As we just concluded, Alice and Bob cannot do better than winning the CHSH game 75% of the time using a deterministic strategy.\n",
+    "But what about a probabilistic strategy?\n",
+    "Could it help Alice and Bob to use randomness — including the possibility of *shared randomness*, where their random choices are correlated?\n",
+    "\n",
+    "It turns out that probabilistic strategies don't help at all to increase the probability that Alice and Bob win.\n",
+    "This is because every probabilistic strategy can alternatively be viewed as a random selection of a deterministic strategy, just like (as was mentioned in the [Single systems](https://learning.quantum.ibm.com/course/basics-of-quantum-information/single-systems) lesson) probabilistic operations can be viewed as random selections of deterministic operations.\n",
```

Comment:
```suggestion
    "This is because every probabilistic strategy can alternatively be viewed as a random selection of a deterministic strategy, just like (as was mentioned in the [Single systems](/learning/courses/basics-of-quantum-information/single-systems/introduction) lesson) probabilistic operations can be viewed as random selections of deterministic operations.\n",
```